### ISSUE TITLE
Update Vulnerable Issues

### ISSUE THREAD

**Issue by thealphadollar** [metakgp/pywikibot#6]
As per github the following dependencies need to be updated and are currently vulnerable:

- Pillow
- requests
- pyopenssl

**Comment by NEERAJAP2001** [comment]
@thealphadollar sir, how can I go ahead with fixing this issue?

**Comment by thealphadollar** [comment]
@NEERAJAP2001 Yes, please take up the issue. Thank you for your enthusiasm and proactiveness to solve the issue <3

**Comment by NEERAJAP2001** [comment]
@thealphadollar thanks sir, but could you please help me with the approach as I'm new to open source.

**Comment by mukul-mehta** [comment]
Hi @NEERAJAP2001 
Thank You for your interest in the project!

Coming to the issue at hand:

- Pywikibot has certain dependencies it needs to run. When working with Python projects, it is common practice to list all of your dependencies inside of a file (Usually named `requirements.txt`). In this project as well, there is such a file where all dependencies and their versions are listed
[This article](https://towardsdatascience.com/virtual-environments-104c62d48c54) describes virtual environments and how to use a `requirements.txt` file

- Many of these dependencies have security flaws in them which can be exploited resulting in unwanted behavior. The community keeps track of such vulnerabilities and developers release newer versions of their libraries with the issues fixed

- In our case, the 3 listed dependencies have security vulnerabilities and we would like to make sure that our code isn't vulnerable, which is why we need to upgrade our versions of these libraries

- To do that, you'll have to figure out the correct versions of the packages (With the issues fixed) for our Python version. Once you know which versions to use, you should update the `requirements.txt` with correct versions of those libraries. But it is also important to make sure that those changes don't break any other changes in the project. Once you upgrade dependencies, it is crucial to test locally your changes and see that everything runs fine. For this project, there a number of unit tests designed for that very purpose. More information about the tests is documented [Here](https://github.com/metakgp/pywikibot/tree/master/tests) and [Here](https://www.mediawiki.org/wiki/Manual:Pywikibot/Test_coverage)


Please ask any questions you have or any issues you face during the process. We'd be happy to help :smile:

**Comment by thealphadollar** [comment]
Oh! @mukul-mehta has beautifully outlined everything that I need to say a word. Here is a screenshot that might help you in updating Pillow.

![image](https://user-images.githubusercontent.com/32812320/92994750-92426100-f51a-11ea-94d7-b2840e90634a.png)
